Flaky test "0-length response with status code: 200 and content type: "

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
If the matter is security sensitive, please disclose it privately to a security contact: https://github.com/kudobuilder/kudo/blob/master/SECURITY_CONTACTS
-->


**What happened**:

We saw a test failure: https://prow.kudo.dev/view/gcs/kudo-prow-logs/pr-logs/pull/kudo/676/test/1156956682320875520

```
--- FAIL: kudo/harness/instance-controller-upgrade (0.01s)
logger.go:37: 15:58:41 | instance-controller-upgrade | Creating namespace: kudo-test-master-crawdad
case.go:84: 0-length response with status code: 200 and content type:
```

I think it probably makes sense to tackle this one at the same time that I do #639

```
        --- FAIL: kudo/harness/parse-param-operator-in-operator (0.01s)
            logger.go:37: 16:58:43 | parse-param-operator-in-operator | Creating namespace: kudo-test-intimate-egret
            case.go:84: 0-length response with status code: 200 and content type: 
```

Here too: https://circleci.com/workflow-run/3420104a-9978-44b4-ba3a-e241a82e95c1

--- a/build/Dockerfile
+++ b/build/Dockerfile
@@ -1,15 +1,15 @@
-ARG GOLANG_VERSION=1.10
+ARG GOLANG_VERSION=1.13
 FROM golang:${GOLANG_VERSION}
 
 
 # Download and install Kubebuilder
 
 # download the release
-RUN curl -L -O https://github.com/kubernetes-sigs/kubebuilder/releases/download/v1.0.8/kubebuilder_1.0.8_linux_amd64.tar.gz
+RUN curl -L -O https://github.com/kubernetes-sigs/kubebuilder/releases/download/v2.1.0/kubebuilder_2.1.0_linux_amd64.tar.gz
 
 # extract the archive
-RUN tar -zxvf kubebuilder_1.0.8_linux_amd64.tar.gz
-RUN mv kubebuilder_1.0.8_linux_amd64 kubebuilder && mv kubebuilder /usr/local/
+RUN tar -zxvf kubebuilder_2.1.0_linux_amd64.tar.gz
+RUN mv kubebuilder_2.1.0_linux_amd64 kubebuilder && mv kubebuilder /usr/local/
 
 # update your PATH to include /usr/local/kubebuilder/bin
 ENV PATH $PATH:/usr/local/kubebuilder/bin

--- a/pkg/test/test_data/create-or-update/00-assert.yaml
+++ b/pkg/test/test_data/create-or-update/00-assert.yaml
@@ -7,12 +7,11 @@ status:
     kind: ProwJob
   # InitialNamesAccepted will change from False -> True once the CRD is ready.
   conditions:
-  - message: no conflicts found
-    reason: NoConflicts
+  - type: NonStructuralSchema
+  - reason: NoConflicts
     status: "True"
     type: NamesAccepted
-  - message: the initial names have been accepted
-    reason: InitialNamesAccepted
+  - reason: InitialNamesAccepted
     status: "True"
     type: Established
   storedVersions:

